src/utility/BlockchainHelper.js

```diff
@@ -370,3 +430,136 @@ const getRFAIContractAddress = () => {
 const getTokenContractAddress = () => {
   return tokenNetworks[process.env.REACT_APP_ETH_NETWORK].address;
 };
+
+const getRFAIContractInstance = () => {
+  const rfaiContractAddress = getRFAIContractAddress();
+
+  try {
+    const ethereum = window.ethereum;
+
+    const web3 = new Web3(ethereum);
+    const rfaiInstance = new web3.eth.Contract(rfaiABI, rfaiContractAddress);
+
+    return rfaiInstance;
+  } catch (error) {
+    throw error;
+  }
+};
+
+// ******************* User Token Balance Functions ***********************
+export const getTokenBalance = async metamaskDetails => {
+  let tokenBalance = 0;
+  const tokenContractAddress = getTokenContractAddress();
+  const accountAddress = metamaskDetails.account;
+
+  if (metamaskDetails.isTxnsAllowed) {
+    const web3 = new Web3(process.env.REACT_APP_INFURA_ENDPOINT);
+    const tokenInstance = new web3.eth.Contract(tokenABI, tokenContractAddress);
+
+    const result = await tokenInstance.methods.balanceOf(accountAddress).call();
+
+    tokenBalance = result.toString();
+
+    return tokenBalance;
+  } else {
+    return tokenBalance.toString();
+  }
+};
+
+// ********************* Fetching The the Token Allowance *******************
+export const getTokenAllowance = async metamaskDetails => {
+  let tokenAllowance = 0;
+  const tokenContractAddress = getTokenContractAddress();
+  const rfaiContractAddress = getRFAIContractAddress();
+  const accountAddress = metamaskDetails.account;
+
+  if (metamaskDetails.isTxnsAllowed) {
+    const web3 = new Web3(process.env.REACT_APP_INFURA_ENDPOINT);
+    const tokenInstance = new web3.eth.Contract(tokenABI, tokenContractAddress);
+
+    const result = await tokenInstance.methods.allowance(accountAddress, rfaiContractAddress).call();
+
+    tokenAllowance = result.toString();
+
+    return tokenAllowance;
+  } else {
+    return tokenAllowance.toString();
+  }
+};
+
+// Fetching The RFAI Token Balance
+export const getRFAITokenBalance = async metamaskDetails => {
+  let rfaiTokenBalance = 0;
+  const rfaiContractAddress = getRFAIContractAddress();
+  const accountAddress = metamaskDetails.account;
+
+  if (metamaskDetails.isTxnsAllowed) {
+    const web3 = new Web3(process.env.REACT_APP_INFURA_ENDPOINT);
```

Comment:
Web3 is initialised in many methods in this file. Why can’t we create a class and assign web3 to a property ? @sridhar

Reply:
The reason is if the metamask is closed by the user the web3 will become invalid. We need to recreate again. Instead on demand creating the object to consider the current instance of metamask.